pkg/grafana/utils.go

```diff
@@ -1,29 +1,44 @@
 package grafana
 
 import (
+	"encoding/json"
 	"regexp"
+
+	gclient "github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grafana-openapi-client-go/models"
 )
 
 var folderURLRegex = regexp.MustCompile("/dashboards/f/([^/]+)")
 
-const generalFolderId = 0
-const generalFolderUID = "general"
-
-func extractFolderUID(d DashboardWrapper) string {
+func extractFolderUID(client *gclient.GrafanaHTTPAPI, d models.DashboardFullWithMeta) string {
 	folderUid := d.Meta.FolderUID
 	if folderUid == "" {
 		urlPaths := folderURLRegex.FindStringSubmatch(d.Meta.FolderURL)
 		if len(urlPaths) == 0 {
-			if d.FolderID == generalFolderId {
+			if d.Meta.FolderID == generalFolderId {
 				return generalFolderUID
 			}
-			folder, err := getFolderById(d.FolderID)
+			folder, err := getFolderById(client, d.Meta.FolderID)
 			if err != nil {
 				return ""
 			}
-			return folder["uid"].(string)
+			return folder.UID
 		}
 		folderUid = urlPaths[1]
 	}
 	return folderUid
 }
+
+func structToMap(s interface{}) (map[string]interface{}, error) {
+	jsonData, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+	if err := json.Unmarshal(jsonData, &result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
```

Comment:
as said above, can we use github.com/mitchellh/mapstructure for this?

Reply:
Ditto